lib/signon/client.rb

```diff
@@ -59,6 +59,18 @@ def create_application(name:, description:, home_uri:, permissions:, redirect_ur
       JSON.parse(res.body)
     end
 
+    def update_application(id:, name:, description:, permissions:)
+      req = build_patch_request("/applications/#{id}", {
+        name: name,
+        description: description,
+        permissions: permissions,
+      })
+      res = do_request(req)
+      raise ApplicationNotFound, "Status: #{res.code}; #{res.message}; #{res.body}" unless res.code == "200"
```

Comment:
Raising `ApplicationNotFound` if `res.code != "200"` may not be appropriate here - perhaps we can raise this if it's a 404, otherwise raise a `ApplicationNotUpdated` or something 🤔

Reply:
Yup - makes sense! I've added error handling for ApplicationNotFound for 404s and ApplicationNotUpdated for everything else.